### PR TITLE
Coverage without diff

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

**Comment by jelly** [comment]
Let's see if the Github comments still work, then drop the commit and merge (if approved).

**Comment by jelly** [comment]
```
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/common/run-tests", line 573, in <module>
    sys.exit(main())
             ^^^^^^
  File "/work/bots/make-checkout-workdir/test/common/run-tests", line 568, in main
    return run(opts, image)
           ^^^^^^^^^^^^^^^^
  File "/work/bots/make-checkout-workdir/test/common/run-tests", line 489, in run
    create_coverage_report()
  File "/work/bots/make-checkout-workdir/test/common/lcov.py", line 476, in create_coverage_report
    comments = get_review_comments(diff_file)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/work/bots/make-checkout-workdir/test/common/lcov.py", line 422, in get_review_comments
    with open(diff_info_file) as f:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/work/bots/make-checkout-workdir/lcov/diff.info'
```

File: pkg/apps/apps.jsx [hunk 1]
```diff
@@ -52,6 +52,10 @@ const App = () => {
     if (!metainfo_db.ready)
         return <EmptyStatePanel loading />;
 
+    if (path.length > 1) {
+        console.log("this is a long path to walk");
```
Comment (cockpituous) [inline pkg/apps/apps.jsx:56]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19081-20230706-152338-650ca044-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: test/common/lcov.py [hunk 2]
```diff
@@ -460,7 +460,7 @@ def create_coverage_report():
         subprocess.check_call(["lcov", "--quiet", "--output", all_file,
                                *itertools.chain(*[["--add", f] for f in lcov_files])])
         subprocess.check_call(["lcov", "--quiet", "--output", diff_file,
-                               "--extract", all_file, "*/github-pr.diff"])
+                               "--extract", all_file, f"{BASE_DIR}/lcov/github-pr.diff"])
```
Comment (mvollmer) [inline test/common/lcov.py]: Why is this necessary?

**Review by mvollmer — APPROVED** [review]
Looks good! Thanks!

File: test/common/lcov.py (hunk 2 above)
Comment (jelly) [reply test/common/lcov.py]: I think there is only one `github-pr.diff` per run right, so this seemed to make sense. Not required.

**Review by mvollmer — CHANGES_REQUESTED** [review]
Hmm, on second thought, this means that the diff will never be part of the coverage report when run locally, no?  I find that useful sometimes. Would it be enough to move the call to get_review_comments into the `rev and pull` section and the leave the rest where it is?

**Comment by mvollmer** [comment]
Hmm, the only way I can make this fail is with a unclean lcov/ directory that already contains som `.info.gz` files with outdated references to github-pr.diff. We should probably clean out `lcov/` in `prepare_for_code_coverage`.  This breaks incremental code coverage, but that is a very manual process anyway, and people shouldn't be using `run-tests` for that.

**Comment by mvollmer** [comment]
@jelly, could you try #19084?

**Comment by martinpitt** [comment]
Moving to draft for now, to get it out of the review queue. This still needs some discussion, and deciding beween this and PR #19084
